### PR TITLE
GN-4258: Number variable - support min/max

### PR DESCRIPTION
**Pull request by dkozickis**
https://binnenland.atlassian.net/browse/GN-4258

https://github.com/lblod/ember-rdfa-editor-lblod-plugins/assets/769698/71bcb49b-3b60-4742-a209-c1387ca44421

- Tests (lint) currently failing because of failure in upstream PR #142

### CODE REVIEW

File: addon/components/variable-plugin/number-settings.hbs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      <AuInput
+        id={{id}}
+        @value={{@minimumValue}}
+        placeholder={{t 'variable-plugin.number.minimum-placeholder'}}
+        @width='block'
+        @type='number'
+      />
+    {{/let}}
+  </AuFormRow>
+  <div class='to'><span>tot</span></div>
```
Comment (lagartoverde) [inline addon/components/variable-plugin/number-settings.hbs:16]: tot should also be translated :)

File: addon/plugins/variable-plugin/utils/constants.ts [hunk 2]
```diff
@@ -92,23 +94,25 @@ export const DEFAULT_VARIABLE_TYPES: Record<string, VariableType> = {
...
         schema.node('placeholder', {
-          placeholderText: selectedCodelist?.label,
+          placeholderText: codelist?.label,
         })
       );
     },
   },
 };
+
+export const EXTRA_ATTRIBUTES_KEY = 'extraAttributes';
```
Comment (lagartoverde) [inline addon/plugins/variable-plugin/utils/constants.ts:118]: maybe the extra attributes should be written to the html in the data- attributes

**Review by lagartoverde — CHANGES_REQUESTED** [review]
Looks good to me overall, just left some minor comments, I don't like the name extra attributes but I don't have a better one, I just think we should use the data- atttributes instead of writting it as extraattributes in the exported HTML

**Comment by x-m-el** [comment]
I'm not sure if I'm looking at the code wrong, but why is "EXTRA_ATTRIBUTES_KEY" needed?
I know the whole variable structure is a bit of a mess (see [GN-4346](https://binnenland.atlassian.net/browse/GN-4346) ), and this feels like trying to maybe partially fix this, but it doesn't feel like the correct way.

I think it'd be better to add the min and max as separate attributes (if using `data-` is a convention for this stuff, could be called data-min and data-max).
This will of course only be used by a number variable, but it will just be empty (and not used) by other variables, which I think is fine.

**Review by x-m-el — COMMENTED** [review]
I'm not sure if I'm looking at the code wrong, but why is "EXTRA_ATTRIBUTES_KEY" needed?
I know the whole variable structure is a bit of a mess (see [GN-4346](https://binnenland.atlassian.net/browse/GN-4346) ), and this feels like trying to maybe partially fix this, but it doesn't feel like the correct way.

I think it'd be better to add the min and max as separate attributes (if using `data-` is a convention for this stuff, could be called `data-min` and `data-max`).
This will of course only be used by a number variable, but it will just be empty (and not used) by other variables, which I think is fine (not the best, but again, this is because it needs to be split up in the future anyway).

**Comment by lagartoverde** [comment]
Looks good to me, but still I would rather it be written as `data-minimumvalue` instead of `minimumvalue`

**Review by lagartoverde — APPROVED** [review]

File: addon/plugins/variable-plugin/nodes.ts [hunk 3]
```diff
@@ -107,29 +160,44 @@ const emberNodeConfig: EmberNodeConfig = {
...
-          const datatype = [...node.children]
+
+          const datatype = nodeChildren
             .find((el) => hasRDFaAttribute(el, 'property', EXT('content')))
             ?.getAttribute('datatype');
-          if (!mappingResource) {
-            return false;
-          }
+
+          const label = node.getAttribute('dataLabel');
```
Comment (elpoelma) [inline addon/plugins/variable-plugin/nodes.ts:199]: This should be `node.getAttribute('data-label');`. Alternatively, you can also use `node.dataset.label`

**Review by elpoelma — CHANGES_REQUESTED** [review]
Looks good overall, but still noticed some issues

**Pull request merged**